controllers/openstackcluster_controller.go

```diff
@@ -291,6 +291,21 @@ func (r *OpenStackClusterReconciler) reconcileNormal(ctx context.Context, log lo
 		if az.ZoneName == "internal" {
 			continue
 		}
+		// If Az given, then check whether it's in the allow list
+		// If no Az given, then by default put into allow list
+		if len(openStackCluster.Spec.AvailabilityZones) > 0 {
```

Comment:
Looks good. I personally would have moved the code to a contains function (`contains(strs []string, str string)`)

Reply:
EDIT: Looks like gocyclo agrees with me :)

Reply:
let's hold this so that we can use .ControlPlane flag

Reply:
Okay so referring what I said in the issue. Let's first move the contains stuff in a separate function. Depending on if the az can be found in the cluster az list you then can set the ControlPlane boolean in line 310 to true or false